File: 204. Count Primes.js
/**
 * @param {number} n
 * @return {number}
 */
var countPrimes = function(n) {
    if (n === 0 || n === 1) return 0;
    var T = [];
    for (var i = 0; i < n; i++)
    {
        if (i === 0 || i === 1)
            T.push(0);
        else
            T.push(1);
    }
    for (i = 2; i*i <= n; i++)
    {
        for (var j = i; j < n; j += i)
        {
            if (T[j] == 1 && i != j)
                T[j] = 0;
        }
    }
    var res = 0;
    for (i = 0; i < n; i++)
        res += T[i];
    return res;
};

File: 204. Count Primes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0 or n == 1:
            return 0
        T = [1 for i in xrange(n)]
        T[0] = 0
        T[1] = 0
        for i in xrange(2, n):
            if i*i <= n:
                for j in xrange(i,n,i): # mark all multiples of i as 0 
                    if T[j] == 1 and j != i:
                        T[j] = 0
        return sum(T) # the numbers with T[number] == 1 are primes
